### PR TITLE
Fixed issue in impl of InMemoryClassLoader.getResource(String)

### PR DESCRIPTION
**Pull request by cdietrich**
Fixed issue in impl of InMemoryClassLoader.getResource(String) preventing from name shadowing for types working correct #166

Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.java.tests/src/org/eclipse/xtext/java/tests/JavaSourceLanguageTest.xtend [hunk 1]
```diff
@@ -85,6 +85,24 @@ class JavaSourceLanguageTest {
         Assert.assertSame(annotation, annotationRef.annotation)
     }
 
+    @Test def void testClassShadowing() {
+    	    val rs = resourceSet('org/eclipse/xtext/java/tests/MySuperClass.java' -> '''
```
Comment (svenefftinge) [inline org.eclipse.xtext.java.tests/src/org/eclipse/xtext/java/tests/JavaSourceLanguageTest.xtend]: Why don't the names of the classes not match the file names?

File: org.eclipse.xtext.java.tests/src/org/eclipse/xtext/java/tests/JavaSourceLanguageTest.xtend [hunk 2]
```diff
@@ -85,6 +85,24 @@ class JavaSourceLanguageTest {
...
+            package org.eclipse.xtext.java.tests;
+            public class MySuperClass2 {
+            	    public void doSomething() {
```
Comment (svenefftinge) [inline org.eclipse.xtext.java.tests/src/org/eclipse/xtext/java/tests/JavaSourceLanguageTest.xtend]: missing closing bracket

**Comment by svenefftinge** [comment]
thanks, for looking into it.

File: org.eclipse.xtext.java.tests/src/org/eclipse/xtext/java/tests/JavaSourceLanguageTest.xtend (hunk 2 above)
Comment (cdietrich) [inline org.eclipse.xtext.java.tests/src/org/eclipse/xtext/java/tests/JavaSourceLanguageTest.xtend]: fixed

File: org.eclipse.xtext.java.tests/src/org/eclipse/xtext/java/tests/JavaSourceLanguageTest.xtend (hunk 1 above)
Comment (cdietrich) [inline org.eclipse.xtext.java.tests/src/org/eclipse/xtext/java/tests/JavaSourceLanguageTest.xtend]: cause copy & paste

**Comment by cdietrich** [comment]
@svenefftinge fixed the naming

**Pull request merged**